katharsis-core/src/main/java/io/katharsis/resource/registry/responseRepository/RelationshipRepositoryAdapter.java

```diff
@@ -1,134 +1,230 @@
+    	RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
+
+			@Override
+			protected JsonApiResponse invoke(RepositoryFilterContext context) {
+				RepositoryRequestSpec request = context.getRequest();
+				Serializable sourceId = request.getId();
+				String fieldName = request.getRelationshipField();
+				QueryAdapter queryAdapter = request.getQueryAdapter();
+				
+		    	Object resource;
+		        if (isAnnotated) {
+		            resource = ((AnnotatedRelationshipRepositoryAdapter) relationshipRepository)
+		                .findOneTarget(sourceId, fieldName, queryAdapter);
+		        } else if(relationshipRepository instanceof QuerySpecRelationshipRepository){
+		        	QuerySpecRelationshipRepository querySpecRepository = (QuerySpecRelationshipRepository) relationshipRepository;
+		        	Class<?> targetResourceClass = querySpecRepository.getTargetResourceClass();
+		        	resource = querySpecRepository.findOneTarget(sourceId, fieldName, request.getQuerySpec(targetResourceClass));
+		        } else {
+		            resource = ((RelationshipRepository) relationshipRepository)
+		                .findOneTarget(sourceId, fieldName, request.getQueryParams());
+		        }
+		        return getResponse(relationshipRepository, resource, request);
+			};
+    	};
+    	RepositoryRequestSpec requestSpec = RepositoryRequestSpecImpl.forFindTarget(moduleRegistry.getResourceRegistry(), queryAdapter, Arrays.asList(sourceId), fieldName, resourceInformation.getResourceClass());
+		return chain.doFilter(newRepositoryFilterContext(requestSpec));
     }
 
 	@SuppressWarnings("rawtypes")
-	public JsonApiResponse findManyTargets(T_ID sourceId, String fieldName, QueryAdapter queryAdapter) {
-        Object resources;
-        if (isAnnotated) {
-            resources = ((AnnotatedRelationshipRepositoryAdapter) relationshipRepository)
-                .findManyTargets(sourceId, fieldName, queryAdapter);
-        }else if(relationshipRepository instanceof QuerySpecRelationshipRepository){
-        	QuerySpecRelationshipRepository querySpecRepository = (QuerySpecRelationshipRepository) relationshipRepository;
-        	Class<?> targetResourceClass = querySpecRepository.getTargetResourceClass();
-            	resources = querySpecRepository.findManyTargets(sourceId, fieldName, toQuerySpec(queryAdapter, targetResourceClass));
-        } else {
-            resources = ((RelationshipRepository) relationshipRepository)
-                .findManyTargets(sourceId, fieldName, toQueryParams(queryAdapter));
-        }
-        RequestSpec requestSpec = new RequestSpec(queryAdapter, sourceId, fieldName, resourceInformation.getResourceClass());
-        return getResponse(relationshipRepository, resources, requestSpec);
+	public JsonApiResponse findManyTargets(I sourceId, String fieldName, QueryAdapter queryAdapter) {
+		RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1188)